src/hal/time.cpp

```diff
@@ -26,37 +26,37 @@ void OswHal::updateTimeProvider() {
         OSW_LOG_D("No provider for Time is available!");
 }
 
-void OswHal::getUTCTime(uint32_t* hour, uint32_t* minute, uint32_t* second) {
+void OswHal::getUTCTime(OSW_TIME* oswTime) {
     RtcDateTime d = RtcDateTime();
     d.InitWithUnix32Time(this->getUTCTime());
-    *hour = d.Hour();
-    *minute = d.Minute();
-    *second = d.Second();
+    oswTime->hour = d.Hour();
+    oswTime->minute = d.Minute();
+    oswTime->second = d.Second();
 }
 
-void OswHal::getTime(time_t& offset, uint32_t* hour, uint32_t* minute, uint32_t* second, bool* afterNoon) {
+void OswHal::getTime(time_t& offset, OSW_TIME* oswTime) {
     RtcDateTime d = RtcDateTime();
     d.InitWithUnix32Time(this->getTime(offset));
     if (!OswConfigAllKeys::timeFormat.get()) {
         if (d.Hour() > 12) {
-            *hour = d.Hour() - 12;
-            if (afterNoon != nullptr) *afterNoon = true;
+            oswTime->hour = d.Hour() - 12;
+            oswTime->afterNoon = true;
```

Comment:
You removed the `if` here - but as the parameter is a pointer, it may be optional... Either change the pointer to force existence (`&`) or condition the assignment with an `if`, please.

Reply:
Now, it is not pointer and neeed param type not optional.